File: solutions/go/raindrops/1/raindrops.go
package raindrops

import "strconv"

func Convert(number int) (str string) {
	if number%3 == 0 {
		str += "Pling"
	}
	if number%5 == 0 {
		str += "Plang"
	}
	if number%7 == 0 {
		str += "Plong"
	}

	if str == "" {
		return strconv.Itoa(number)
	}

	return str
}

File: solutions/ruby/raindrops/1/raindrops.rb
# frozen_string_literal: true

# Converts a number into its corresponding raindrop sounds
module Raindrops
  def self.convert(number)
    str = ''
    str += 'Pling' if (number % 3).zero?
    str += 'Plang' if (number % 5).zero?
    str += 'Plong' if (number % 7).zero?

    str.empty? ? number.to_s : str
  end
end
